packages/block-editor/src/components/block-context/README.md

```diff
@@ -0,0 +1,57 @@
+Block Context
+=============
+
+Block Context is a React implementation of WordPress's block context. Block context, much like [React's context](https://reactjs.org/docs/context.html), is a method for passing and inheriting values deeply through a hierarchy of blocks. Because of the similarities with React's context, the client-side implementation here is quite minimal. It is complemented by equivalent behaviors in the server-side rendering of a block.
+
+Note that the implementation of Block Context is distinct from [the `BlockEdit` context](../block-edit). While it is true that both provide context relevant for the editing of a block, Block Context is implemented separately so as to prioritize it as most identifiable amongst the machinery of block context, and not amongst other client-side editing context of a block.
+
+## Usage
+
+Currently, only the [Provider component](https://reactjs.org/docs/context.html#contextprovider) is made availble on the public interface of the `@wordpress/block-editor` module. This can be used to add or override context which can then be consumed by blocks rendered within that context.
```

Comment:
> Might be worth clarifying that it's the blocks' _editing interfaces_ which consume the context here.

Yep. I might normally be somewhat reluctant to hard-code references to implementations as they happen to exist today (given the "Open Question" about making this available to `save`), but I think it makes sense to phrase it this way, especially considering it might help to avoid potential confusion.

Reply:
> Might be worth clarifying that it's the blocks' _editing interfaces_ which consume the context here.

Addressed in cceb12e09.